packages/workflow/src/Extensions/ArrayExtensions.ts

```diff
@@ -344,12 +305,14 @@ function intersection(value: unknown[], extraArgs: unknown[][]): unknown[] {
 	}
 	const newArr: unknown[] = [];
 	for (const v of value) {
-		if (others.findIndex((w) => deepCompare(w, v)) !== -1) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-call
+		if (others.findIndex((w) => deepEqual(w, v)) !== -1) {
```

Comment:
Maybe this is a product requirement but IMO we should pass `strict: true` as config options to `deepEqual` to prevent it from typecasting when comparing objects. Without strict we'd get result like these:
<img width="684" alt="CleanShot 2023-02-14 at 12 39 14@2x" src="https://user-images.githubusercontent.com/12657221/218729333-5c6fb1a7-58ed-4f80-8796-2b0de578c7bb.png">

Reply:
Yeah, that's not expected.